### PR TITLE
Threaded Pipes: Adjust Ropsten to better deal with nonce issues

### PR DESCRIPTION
**Pull request by Shadowfiend**
From the main commit:

> Because of how HDWalletProvider works, nonces are cached by a local
> provider, which also affects calls to `getTransactionCount`. This means
> if for some reason a bad nonce is cached, it's impossible to recover
> without kicking the function altogether.
>
> Instead, the grant process has been reworked to recursively invoke
> itself with a higher nonce whenever a `nonce too low` or `replacement
> transaction underpriced` error is reported.
>
> Also cleaned up error handling, success handling, and separated request
> and response handling from the grant issuance itself.

A few other small adjustments are also bundled in here.

### CODE REVIEW

File: infrastructure/gcp/keep-test/google-functions/keep-faucet/issue-grant.js [hunk 1]
```diff
@@ -53,11 +57,53 @@ const permissiveStakingPolicyAddress =
...
+  },
+}
+const SUCCESSES = {
+  created: `
+      Created token grant with {grantAmount} KEEP for account: {granteeAccount}\n
+      You can follow the transaction at https://ropsten.etherscan.io/tx/{transactionHash}\n
+      You can manage your token grants at: https://dashboard.test.keep.network .\n
+      You can find us on Discord at: https://discord.gg/jqxBU4m .\n
+  `,
+}
```
Comment (pdyraga) [inline infrastructure/gcp/keep-test/google-functions/keep-faucet/issue-grant.js:93]: This is awesome. 👏

**Review by pdyraga — COMMENTED** [review]
The code looks good (awesome!), before merging I want to test it but Ropsten gives me no chance now.

**Comment by Shadowfiend** [comment]
Fwiw this is currently deployed for the Ropsten faucet 😬 But yeah, give it a test whirl locally when you get a chance—`npm run start` does the trick to get you a local setup, but you'll need the environment variables set up appropriately.

**Comment by Shadowfiend** [comment]
Pushed a truffle-hdwalletprovider version bump here in response to [an issue reported on Discord](https://discordapp.com/channels/590951101600235531/686646403057057968/749984625958584401) where the provider wasn't reconnecting when the WebSocket connection broke.

**Review by pdyraga — APPROVED** [review]
Tested locally, works as advertised 👍

**Pull request merged**